include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -1246,6 +1246,173 @@ namespace Differentiation
       const std::vector<std::pair<ExpressionType, ValueType>> &symbol_values,
       const bool force_cyclic_dependency_resolution = false);
 
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * symbolic expression.
+     * The symbols in the @p expression that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     *
+     * @note This function call is typically expensive, as by default it performs a
+     * dictionary substitution for the symbols in the symbolic expression.
+     * Should the numerical values of some symbolic expression be derired, then
+     * this performance deficit may be mitigated through the use of the
+     * BatchOptimizer class.
+     * Situation dependent, the overhead of using a typical dictionary based
+     * substitution may be on par with that of the a substitution performed
+     * using a BatchOptimizer. This is because there is an overhead to setting
+     * up the optimizer, so this should be taken into consideration if
+     * substitution is to occur for the given symbolic
+     * expression only a few times.
+     *
+     * @note If the symbols stored in the map are explicitly dependent on one another,
+     * then the returned result depends on order in which the map is traversed.
+     * It is recommended to first resolve all inter-dependencies in the map
+     * using the resolve_explicit_dependencies() function.
+     * Examples:
+     *   1. If <tt>map["a"] == 1</tt> and <tt>map["b"] == "a"+ 2</tt> then
+     *     then the function $f(a,b(a)) = a+b$ will be evaluated and the result
+     *     $f\vert_{a=1,b=a+2} = 3+a$ is returned. This return is because the
+     *     symbol "a" is substituted throughout the function first, and only
+     * then is the symbol "b(a)" substituted, by which time its explicit
+     * dependency on "a" cannot be resolved.
+     *  2. If <tt>map["a"] == "b"+2</tt> and <tt>map["b"] == 1</tt> then
+     *     then the function $f(a(b),a) = a+b$ will be evaluated and the result
```

Comment:
```suggestion
     *     then the function $f(a(b),b): = a+b$ will be evaluated and the result
```
or just document `f(a,b):=a+b` above.